### PR TITLE
Move string functions from tlibc to libk

### PR DESCRIPTION
**Pull request by Herbstein**

### CODE REVIEW

File: kernel/libk/kmem.c [hunk 1]
```diff
@@ -45,7 +45,6 @@ void *krealloc(void *ptr, size_t bytes) {
         return ptr;
     } else {
         void *new = kmalloc(bytes);
-        memcpy(new, ptr, ptr_md->size);
```
Comment (Herbstein) [inline kernel/libk/kmem.c]: I think there needs to be a `kmemcpy` added in the new file.

**Comment by iankronquist** [comment]
The release versions of vagrant and virtualbox aren't compatible right now, so I can't test it until I can run vagrant. I'll check back tomorrow.

**Comment by iankronquist** [comment]
Hm this fails for me when I run the userspace test suite:

```
iankronquist(kernel-of-truth)(Herbstein-tlibc_reformat) → make tests
clang kernel/libk/tests/stubs_tests.c kernel/libk/tests/kmem_tests.c  -o build/tests/kmem -std=c11 -O0 -Wall -Wextra -g -I ./include -coverage -Wno-format -D ARCH_USERLAND
In file included from kernel/libk/tests/kmem_tests.c:5:
In file included from ./include/libk/kmem.h:5:
./include/libk/kmemcpy.h:5:7: error: expected parameter declarator
void* memcpy(void*, const void*, size_t);
      ^
/usr/include/secure/_string.h:65:43: note: expanded from macro 'memcpy'
  __builtin___memcpy_chk (dest, src, len, __darwin_obsz0 (dest))
                                          ^
/usr/include/secure/_common.h:38:63: note: expanded from macro '__darwin_obsz0'
#define __darwin_obsz0(object) __builtin_object_size (object, 0)
                                                              ^
In file included from kernel/libk/tests/kmem_tests.c:5:
In file included from ./include/libk/kmem.h:5:
./include/libk/kmemcpy.h:5:7: error: expected ')'
/usr/include/secure/_string.h:65:43: note: expanded from macro 'memcpy'
  __builtin___memcpy_chk (dest, src, len, __darwin_obsz0 (dest))
                                          ^
/usr/include/secure/_common.h:38:63: note: expanded from macro '__darwin_obsz0'
#define __darwin_obsz0(object) __builtin_object_size (object, 0)
                                                              ^
./include/libk/kmemcpy.h:5:7: note: to match this '('
/usr/include/secure/_string.h:65:43: note: expanded from macro 'memcpy'
  __builtin___memcpy_chk (dest, src, len, __darwin_obsz0 (dest))
                                          ^
/usr/include/secure/_common.h:38:54: note: expanded from macro '__darwin_obsz0'
#define __darwin_obsz0(object) __builtin_object_size (object, 0)
                                                     ^
In file included from kernel/libk/tests/kmem_tests.c:5:
In file included from ./include/libk/kmem.h:5:
./include/libk/kmemcpy.h:5:7: warning: type specifier missing, defaults to 'int'
      [-Wimplicit-int]
void* memcpy(void*, const void*, size_t);
      ^
/usr/include/secure/_string.h:65:43: note: expanded from macro 'memcpy'
  __builtin___memcpy_chk (dest, src, len, __darwin_obsz0 (dest))
                                          ^
/usr/include/secure/_common.h:38:32: note: expanded from macro '__darwin_obsz0'
#define __darwin_obsz0(object) __builtin_object_size (object, 0)
                               ^
In file included from kernel/libk/tests/kmem_tests.c:5:
In file included from ./include/libk/kmem.h:5:
./include/libk/kmemcpy.h:5:7: error: conflicting types for
      '__builtin___memcpy_chk'
/usr/include/secure/_string.h:65:3: note: expanded from macro 'memcpy'
  __builtin___memcpy_chk (dest, src, len, __darwin_obsz0 (dest))
  ^
./include/libk/kmemcpy.h:5:7: note: '__builtin___memcpy_chk' is a builtin with
      type 'void *(void *, const void *, unsigned long, unsigned long)'
/usr/include/secure/_string.h:65:3: note: expanded from macro 'memcpy'
  __builtin___memcpy_chk (dest, src, len, __darwin_obsz0 (dest))
  ^
1 warning and 3 errors generated.
make: *** [build/tests/kmem_tests] Error 1

```

File: include/arch/x86/idt.h [hunk 2]
```diff
@@ -3,8 +3,8 @@
 
 #include <stddef.h>
 #include <stdint.h>
-#include <string.h>
 
+//#include <libk/kstring.h>
```
Comment (iankronquist) [inline include/arch/x86/idt.h]: Can you remove this line if it's not necessary?

File: kernel/libk/kmem.c (hunk 1 above)
Comment (iankronquist) [inline kernel/libk/kmem.c]: I'm not sure I follow. This is necessary for `krealloc` to work properly.

**Comment by iankronquist** [comment]
Do you think you could write docstrings for these functions in the header files? They could be fairly straightforward "Kernel implementation of memset" etc.

Here is an example: 
https://github.com/iankronquist/kernel-of-truth/blob/master/include/libk/physical_allocator.h#L44

You might take a look at the generated documentation for an example of what's produced.

**Comment by iankronquist** [comment]
This will have conflicts with #62.

**Comment by Herbstein** [comment]
It seems you have an idea for how we should be proceeding. Do you still think we should try and hamfist this into your new code, or should we do a new refactor when this part of the kernel isn't being changed as often?

**Comment by iankronquist** [comment]
@Herbstein either way works. I just wanted to mention that in case #62 got merged first. However, I don't foresee it being finished until the end of next weekend.

**Comment by iankronquist** [comment]
Also, I would like to thank you for all your help!

**Comment by iankronquist** [comment]
I'm going to go ahead and close this since several other things have been refactored since this was opened and caused merge conflicts. If you'd like to work on this or other things, please let me know!

**Pull request closed without merging**